noggin/controller/password.py

```diff
@@ -91,7 +91,14 @@ def user_settings_password(ipa, username):
     form = PasswordResetForm()
 
     # check if an OTP token exists. If so, the user is using OTP.
-    using_otp = bool(ipa.otptoken_find(ipatokenowner=username))
+    using_otp = bool(
+        ipa.otptoken_find(
+            o_ipatokenowner=username,
+            o_ipatokenotpalgorithm='sha512',
```

Comment:
This line and the next 2 should not be necessary with python_freeipa >= 1.0.6 (see [this PR](https://github.com/opennode/python-freeipa/pull/44)), we can leave it here but could you please add a comment in the code to explain? Thanks.

Reply:
Removing these lines with `python-freeipa` 1.0.6 breaks tests. This is with a reset & repopulated FreeIPA and removed VCR cassettes (which works without the removed lines).

Reply:
I destroyed and re-provisioned my Vagrant box, and now it works. No idea why, though.